src/tap/line.py

```diff
@@ -111,7 +111,17 @@ def __str__(self):
         diagnostics = ""
         if self.diagnostics is not None:
             diagnostics = "\n" + self.diagnostics.rstrip()
-        return f"{is_not}ok {self.number} {self.description}{directive}{diagnostics}"
+        yaml_block = ""
+        if self._yaml_block is not None:
+            indent = " "
+            indented_yaml_block = "\n".join(
+                f"{indent}{line}" for line in self._yaml_block.splitlines()
+            )
+            yaml_block = f"\n{indent}---\n{indented_yaml_block}\n{indent}..."
```

Comment:
Just checking... should there be a trailing newline here? I suspect the answer is no because is probably printed later, but I would hate to mistakenly combine test line like `<end of yaml>...not ok 43 yada yada`
```suggestion
            yaml_block = f"\n{indent}---\n{indented_yaml_block}\n{indent}...\n"
```

Reply:
No newline needed here. in Tracker.generate_tap_report it calls print for each Line object. a Result is still a "Line" (even though it's actually multi-line).
```
    def generate_tap_report(self, test_case, tap_lines, out_file):
        self._write_test_case_header(test_case, out_file)

        for tap_line in tap_lines:
            print(tap_line, file=out_file)
```
Plus for reassurance, you can see that even if there was no yaml block added, diagnostics in this method did not have a trailing newline either